packages/lesswrong/components/users/account/DeactivateAccountSection.tsx

```diff
@@ -0,0 +1,44 @@
+import { Components, registerComponent } from '@/lib/vulcan-lib';
+import React from 'react';
+import { useUpdate } from '@/lib/crud/withUpdate';
+import { useFlashErrors } from '@/components/hooks/useFlashErrors';
+
+const DeactivateAccountSection = ({
+  user,
+}: {
+  user: UsersEdit,
+}) => {
+  const { ActionButtonSection } = Components;
+  const { mutate: rawUpdateUser, loading } = useUpdate({
+    collectionName: "Users",
+    fragmentName: 'UsersEdit',
+  });
+  const updateUser = useFlashErrors(rawUpdateUser);
+
+  const disableDeactivateButton = !!user.permanentDeletionRequestedAt && !!user.deleted;
+
+  return (
+    <ActionButtonSection
+      description="Deactivating your account means your posts and comments will be listed as '[Anonymous]', and your user profile won't accessible. This can be reversed at any time."
+      buttonText={`${user.deleted ? "Reactivate" : "Deactivate"} account`}
+      buttonProps={{ variant: "contained", disabled: disableDeactivateButton }}
+      loading={loading}
+      onClick={() => {
+        void updateUser({
+          selector: { slug: user.slug },
```

Comment:
Probably a nit, but maybe it'd be marginally safer to use `_id` instead of `slug` since `slug` is nullable, mutable and not enforced to be unique at the DB level.